pinot-common/src/main/java/com/linkedin/pinot/common/data/Schema.java

```diff
@@ -94,236 +89,241 @@ public static ZNRecord toZNRecord(Schema schema)
     return record;
   }
 
-  public Schema() {
-    dimensions = new ArrayList<String>();
-    metrics = new ArrayList<String>();
-    dimensionFieldSpecs = new ArrayList<DimensionFieldSpec>();
-    metricFieldSpecs = new ArrayList<MetricFieldSpec>();
-    timeFieldSpec = null;
-    fieldSpecMap = new HashMap<>();
+  public static String toSchemaString(Schema schema)
+      throws IOException {
+    String schemaString = schema.getJSONSchema();
+
+    // Throw out the exception if it happens.
```

Comment:
Yes, and that's what this part of the code checks for.
I will make the comment more clear. When any exception happens in getJSONSchema(), it will return null.